### PR TITLE
Try all host addresses with a 3 seconds timeout.

### LINKED ISSUE #1: lcg-info does not try alternative IP addresses for replicated BDII
From https://ggus.eu/index.php?mode=ticket_info&ticket_id=140478:

lcg-info cannot exploit DNS-based BDII replication (load-balancing,high-availability) since it does not try alternative IP addresses:
```sh
$ host lcg-bdii.cern.ch
lcg-bdii.cern.ch has address 188.184.88.34
lcg-bdii.cern.ch has address 188.184.93.186
lcg-bdii.cern.ch has address 188.185.75.87
lcg-bdii.cern.ch has IPv6 address 2001:1458:d00:8::100:351
lcg-bdii.cern.ch has IPv6 address 2001:1458:201:e4::100:5b4
lcg-bdii.cern.ch has IPv6 address 2001:1458:201:e4::100:1c
$ strace -f lcg-info --list-ce --bdii lcg-bdii.cern.ch:2170 2>&1 |grep connect\(.*2170
connect(4, {sa_family=AF_INET, sin_port=htons(2170), sin_addr=inet_addr("188.184.88.34")}, 16) = -1 EINPROGRESS (Operation now in progress)
connect(4, {sa_family=AF_INET, sin_port=htons(2170), sin_addr=inet_addr("188.184.88.34")}, 16) = -1 ECONNREFUSED (Connection refused)
```

### PR DESCRIPTION
**Pull request by benja-egi**

### CODE REVIEW

**Comment by gwarf** [comment]
@benja-egi can you please rebase? I would expect it should trigger a build in Travis.

**Comment by benja-egi** [comment]
rebased

**Comment by gwarf** [comment]
Cool, looks good!
Just unsure about the timeout duration.
Maybe 3s may not be enough?
Should we provide a default and allow people to override it on command line using a parameter?
Anyone any strong opinion?

**Comment by maarten-litmaath** [comment]
Hi all,
3s is way too short for anything on the Internet...
I suggest 10s instead.

**Review by gwarf — CHANGES_REQUESTED** [review]
Thanks for checking this @benja-egi, please update the default timeout to 10s.
Bonus points if you make it overridable :)

**Comment by benja-egi** [comment]
Now defaults to 10 sec, added --timeout flag for extra configuration and doc.

**Review by gwarf — APPROVED** [review]

File: src/lcg-info [hunk 1]
```diff
@@ -1033,6 +1035,14 @@ VOMS:<fqan>, where <fqan> is a VOMS Fully Qualified Attribute Name. The
 * wildcard is allowed. When used with --list-ce, <vo> and VO:<vo> are
 equivalent.
 
+=item B<--timeout>
+
+Restricts the time waiting for a TCP connection to establish.
+Defaults to 10.0 seconds. Using a shorter timeout might speedup the
+falling back to non-primary addresses when the endpoint implements
+high-availability using multiple address records. On the contrary,
+a longer timeout might help on unrealiable network.
```
Comment (gwarf) [inline src/lcg-info]:
```suggestion
Restricts the time waiting for a TCP connection to establish.
Defaults to 10.0 seconds. Using a shorter timeout might speedup
falling back to alternative addresses when the endpoint implements
high-availability using multiple address records. On the contrary,
a longer timeout might help with unreliable network.
```

**Review by gwarf — CHANGES_REQUESTED** [review]
Just small language changes, thanks a lot!

**Review by gwarf — APPROVED** [review]

**Pull request merged**